core/DataTable/Row.php

```diff
@@ -512,6 +495,51 @@ public function sumRow(Row $rowToSum, $enableCopyMetadata = true, $aggregationOp
         }
     }
 
+    protected function getColumnAggregationOperation($columnNameOrId, $aggregationOperations)
+    {
+        // max_actions is a core metric that is generated in ArchiveProcess_Day. Therefore, it can be
+        // present in any data table and is not part of the $aggregationOperations mechanism.
+        if ($columnNameOrId == Metrics::INDEX_MAX_ACTIONS) {
+            return 'max';
+        }
+
+        if (empty($aggregationOperations) || !is_array($aggregationOperations)) {
+            return 'sum';
+        }
+
+        $operationName = 'sum';
+
+        if (isset($aggregationOperations[$columnNameOrId])) {
+            $operationName = $aggregationOperations[$columnNameOrId];
+        } elseif (is_numeric($columnNameOrId)) {
+            // if the column was provided as index, check if a aggration is set for the real name
+            $metricsIdToName = Metrics::getMappingFromIdToName();
+            if (isset($metricsIdToName[$columnNameOrId]) && isset($aggregationOperations[$metricsIdToName[$columnNameOrId]])) {
+                $operationName = $aggregationOperations[$metricsIdToName[$columnNameOrId]];
+            }
+        } else {
+            // if the column was provided as string, check if a aggration is set for the index
+            $metricsNameToId = Metrics::getMappingFromNameToId();
+            if (isset($metricsNameToId[$columnNameOrId]) && isset($aggregationOperations[$metricsNameToId[$columnNameOrId]])) {
+                $operationName = $aggregationOperations[$metricsNameToId[$columnNameOrId]];
+            }
+        }
```

Comment:
This fixes an issue that in parts already appear before. We are setting aggregation operations for certain columns like here:
https://github.com/matomo-org/matomo/blob/5863dc9267b373997c8b25b2aea79ccbd96bcea8/plugins/Events/API.php#L159-L164

In that specific example the operations are set together with the metric id. That requires the `ReplaceColumnNames` filter **NOT** to run before, as it otherwise can't find the correct operation.

The code above will automatically try to find the correct operation by mapping metric ids to their name or vice versa to see if a provided operation is found then.